File: pios.projekt.API/pios.projekt.models/Models/Teacher.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace pios.projekt.models.Models
{
    public class Teacher
    {
        public int Id { get; set; }
        public string Name { get; set; }
        public string Surname { get; set; }

        public List<Subject> subjects { get; set; } = new List<Subject>();

        public List<SchoolClass> schoolClasses { get; set; } = new List<SchoolClass>();


    }
}

File: pios.projekt.front/src/Models/Teacher.ts
import { SchoolClass } from './SchoolClass';
import { Subject } from './Subject';

export interface Teacher {
    id: number;
    name: string;
    surname: string;
    subjects: Subject[];
    classes: SchoolClass[];
}

export class TeacherClass implements Teacher {
    id = -1;
    name = '';
    surname = '';
    subjects = [];
    classes = [];
}
